src/main/java/de/rwth/idsg/steve/repository/impl/WebUserRepositoryImpl.java

```diff
@@ -0,0 +1,279 @@
+/*
+ * SteVe - SteckdosenVerwaltung - https://github.com/steve-community/steve
+ * Copyright (C) 2013-2024 SteVe Community Team
+ * All Rights Reserved.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+package de.rwth.idsg.steve.repository.impl;
+
+import de.rwth.idsg.steve.SteveException;
+import de.rwth.idsg.steve.repository.WebUserRepository;
+import de.rwth.idsg.steve.repository.dto.WebUser;
+import de.rwth.idsg.steve.web.dto.WebUserForm;
+import de.rwth.idsg.steve.web.dto.WebUserQueryForm;
+import jooq.steve.db.tables.records.WebauthoritiesRecord;
+import jooq.steve.db.tables.records.WebusersRecord;
+import lombok.extern.slf4j.Slf4j;
+import org.jooq.DSLContext;
+import org.jooq.JoinType;
+import org.jooq.Record3;
+import org.jooq.Result;
+import org.jooq.SelectQuery;
+import org.jooq.exception.DataAccessException;
+import org.jooq.impl.DSL;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+import java.util.List;
+
+import static de.rwth.idsg.steve.utils.CustomDSL.includes;
+import static jooq.steve.db.tables.Webusers.WEBUSERS;
+import static jooq.steve.db.tables.Webauthorities.WEBAUTHORITIES;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+
+
+ /**
+ * @author fnkbsi
+ * @since 01.04.2022
+ */
+@Slf4j
+@Repository
+public class WebUserRepositoryImpl implements WebUserRepository {
+
+    @Autowired private DSLContext ctx;
+
+    private PasswordEncoder encoder = new BCryptPasswordEncoder();
```

Comment:
you should use [standard Bean PasswordEncoder](https://github.com/steve-community/steve/blob/master/src/main/java/de/rwth/idsg/steve/config/SecurityConfiguration.java#L77)